File: 1-two-sum/1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        int[][] sorted = new int[n][2];

        for (int i = 0; i < n; ++i) {
            sorted[i][0] = nums[i];
            sorted[i][1] = i;
        }
        Arrays.sort(sorted, Comparator.comparingInt(o -> o[0]));

        int left = 0;
        int right = n - 1;
        while (left < right) {
            int sum = sorted[left][0] + sorted[right][0];
            if (sum < target) {
                ++left;
            } else if (sum > target) {
                --right;
            } else {
                return new int[] {sorted[left][1], sorted[right][1]};
            }
        }
        return new int[] {};
    }
}

File: 1-two-sum/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let sorted = Array.from(nums.entries())
                      .sort((a, b) => a[1] - b[1])
    let left = 0
    let right = nums.length - 1
    while (left < right) {
        let sum = sorted[left][1] + sorted[right][1]
        if (sum < target) ++left
        else if (sum > target) --right
        else return [sorted[left][0], sorted[right][0]]
    }
    return []
};
